src/components/chatbox/message.tsx

```diff
@@ -0,0 +1,317 @@
+			<div className={classNames(s.content)}>
+				<ReactMarkdown
+					className={s.markdown}
+					components={{
+						// @ts-expect-error - ignore this type incompatibility
+						pre: MdxPre,
+						p: (props) => (
+							<Text /* Body/200/Medium */
+								size={200}
+								weight="medium"
+								{...props}
+							/>
+						),
+					}}
+					remarkPlugins={[remarkGfm]}
+					rehypePlugins={[[rehypePrism, { ignoreMissing: true }]]}
+				>
+					{markdown}
+				</ReactMarkdown>
+
+				<div
+					className={classNames(s.assistantMessageFooter, {
+						[s.assistantMessageFooterHidden]: !showActions,
+					})}
+				>
+					<span className={s.divider} />
+					<div className={s.actionButtons}>
+						<Button
+							size="small"
+							color="secondary"
+							icon={
+								isCopied ? (
+									<IconCheckSquare24 height={12} width={12} />
+								) : (
+									<IconClipboard24 height={12} width={12} />
+								)
+							}
+							aria-label="Copy to clipboard"
+							text={isCopied ? 'Copied' : 'Copy'}
+							onClick={handleCopy}
+						/>
+
+						<Button
+							size="small"
+							color="secondary"
+							className={classNames({ [s.ratingLike]: rating == 1 })}
+							disabled={rating == 1}
+							icon={<IconThumbsUp24 height={12} width={12} />}
+							aria-label="Like this response"
+							onClick={async () => {
+								setRating(1) // do an optimistic UI update
+								await handleFeedback({ rating: 1 })
+							}}
+						/>
+
+						<Button
+							size="small"
+							color="secondary"
+							className={classNames({ [s.ratingDislike]: rating == -1 })}
+							disabled={rating == -1}
+							icon={<IconThumbsDown24 height={12} width={12} />}
+							aria-label="Dislike this response"
+							onClick={async () => {
+								setRating(-1) // do an optimistic UI update
+								await handleFeedback({ rating: -1 })
+							}}
+						/>
+					</div>
+
+					<div
+						className={classNames(s.feedbackForm, {
+							[s.feedbackFormVisible]: rating != 0,
+						})}
+					>
+						{rating != 0 ? (
+							<FeedbackForm
+								questions={[
+									{
+										id: 'reason',
+										type: 'text',
+										label: 'Provide additional feedback to help us improve',
+										placeholder:
+											rating == 1
+												? 'What did you like about the response?'
+												: 'What was the issue with the response? How could it be improved?',
+										optional: true,
+										buttonText: 'Submit answer',
+									},
+								]}
+								finishedText={
+									<div>
+										Thank you! Your feedback will help us improve our websites.
+									</div>
+								}
+								onQuestionSubmit={async (responses) => {
+									const value = responses[0].value
+									await handleFeedback({ rating, text: value })
+									return
+								}}
+							/>
+						) : null}
+					</div>
+				</div>
+			</div>
+
+			<div className={classNames(s.gutter)}></div>
+		</div>
+	)
+}
+
+// TODO(kevinwang): error styling.
```

Comment:
🎏 Flagging in case this is a `TODO` that needs doing before merge!

Reply:
Will address this in an upcoming chunk of VQA work!